front/app/community/[siren]/components/FicheMarchesPublics/Treemap.tsx

```diff
@@ -0,0 +1,305 @@
+'use client';
+
+import { useEffect, useRef, useState } from 'react';
+
+import DownloadSelector from '@/app/community/[siren]/components/DownloadDropDown';
+import YearSelector from '@/app/community/[siren]/components/YearSelector';
+import { MarchePublic } from '@/app/models/marche_public';
+import { Switch } from '@/components/ui/switch';
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from '@/components/ui/table';
+import { formatNumber } from '@/utils/utils';
+import * as d3 from 'd3';
+
+import SectorTable from './SectorTable';
+
+type TreeNode = {
+  type: 'node';
+  value: number;
+  name: string;
+  children: Tree[];
+};
+type TreeLeaf = {
+  type: 'leaf';
+  name: string;
+  value: number;
+};
+
+type Tree = TreeNode | TreeLeaf;
+
+function wrapText(text: string, maxWidth: number): string[] {
+  const words = text.split(' ');
+  const lines: string[] = [];
+  let currentLine = words[0] ?? '';
+
+  for (let i = 1; i < words.length; i++) {
+    const word = words[i];
+    const width = (currentLine + ' ' + word).length * 6.5; // Estimation ~6.5px par caractère
+    if (width < maxWidth) {
+      currentLine += ' ' + word;
+    } else {
+      lines.push(currentLine);
+      currentLine = word;
+    }
+  }
+  lines.push(currentLine);
+  return lines;
+}
+
+function generateColorMap(names: string[]): Record<string, string> {
+  const colorMap: Record<string, string> = {};
+  const total = names.length;
+
+  names.forEach((name, index) => {
+    const hue = Math.round((360 / total) * index);
+    colorMap[name] = `hsl(${hue}, 65%, 55%)`;
+  });
+
+  return colorMap;
+}
+
+type YearOption = number | 'All';
+
+function getAvailableYears(data: MarchePublic[]) {
+  return [...new Set(data.map((item) => item.datenotification_annee))].sort(
+    (a: number, b: number) => a - b,
+  );
+}
+
+export default function Treemap({ data }: { data: MarchePublic[] }) {
+  const [tooltip, setTooltip] = useState<{
+    visible: boolean;
+    x: number;
+    y: number;
+    name: string;
+    value: number;
+  }>({
+    visible: false,
+    x: 0,
+    y: 0,
+    name: '',
+    value: 0,
+  });
```

Comment:
Tu peux creer un composnat tooltip ! (voir plus bas) puis utiliser son type de props ici

```suggestion
  const [tooltip, setTooltip] = useState<TooltipProps>({
    visible: false,
    x: 0,
    y: 0,
    name: '',
    value: 0,
  });
```

Reply:
done